atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/DefaultSweepProgressResetter.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.sweep.queue;
+
+import com.palantir.atlasdb.keyvalue.api.KeyValueService;
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.atlasdb.table.description.SweeperStrategy;
+import com.palantir.logsafe.SafeArg;
+import com.palantir.logsafe.logger.SafeLogger;
+import com.palantir.logsafe.logger.SafeLoggerFactory;
+import java.util.Set;
+import java.util.function.Supplier;
+
+public class DefaultSweepProgressResetter implements SweepProgressResetter {
+    private static final SafeLogger log = SafeLoggerFactory.get(DefaultSweepProgressResetter.class);
+    private final KeyValueService keyValueService;
+    private final TableReference bucketProgressTable;
+    private final TableReference sweepBucketAssignedTable;
+    private final ShardProgress progress;
+    private final Supplier<Integer> numberOfShardsSupplier;
+
+    public DefaultSweepProgressResetter(
+            KeyValueService keyValueService,
+            TableReference bucketProgressTable,
+            TableReference sweepBucketAssignedTable,
+            ShardProgress progress,
+            Supplier<Integer> numberOfShardsSupplier) {
+        this.keyValueService = keyValueService;
+        this.bucketProgressTable = bucketProgressTable;
+        this.sweepBucketAssignedTable = sweepBucketAssignedTable;
+        this.progress = progress;
+        this.numberOfShardsSupplier = numberOfShardsSupplier;
+    }
+
+    @Override
+    public void resetProgress(Set<SweeperStrategy> strategies) {
+        log.info("Truncating bucket progress and sweep bucket assigned tables as part of resetting sweep progress");
+        keyValueService.truncateTables(Set.of(bucketProgressTable, sweepBucketAssignedTable));
```

Comment:
not actionable: This includes bucket records too, but that's fine - we will restart from zero when identifying new buckets. It might take a long time to re-identify new buckets, so elsewhere, if progress is 0 we will take a fresh timestamp to identify the first bucket - and otherwise, we will use the min progress for a given shard.